### PR TITLE
Added a simple translation map

### LINKED ISSUE #7: Translate Instructions
Currently the "In 2 kilometers" part is not translated.

Since it's only a handful of strings, for now it might be enough to simply translate them in the code. In the future it would be interesting to see how we can solve this issue in a nice way, maybe reusing some of the translation components of GH core.

### PR DESCRIPTION
**Pull request by boldtrn**
Fixes #7.

This PR adds a simple translation map. Nothing fancy. I added translations for de and en. If you pass an unknown translation, it will fall back to en.

### CODE REVIEW

**Comment by karussell** [comment]
Instead of a new class and new mechanism couldn't we just use the existing TranslationMap stuff and add certain translations?

**Comment by boldtrn** [comment]
> Instead of a new class and new mechanism couldn't we just use the existing TranslationMap stuff and add certain translations?

Good question. The thing is, the initialization of the TranslationMap happens outside of the scope of this library, this is done e.g. in the gh-web module. I don't think that there is currently a way of adding more translation after they have been initialized?

We could just add these 3 translations to the official excel file. This would be certainly the easiest option.

**Comment by karussell** [comment]
If it is not possible to use the same instance, we can still use the same classes avoiding rewriting later and adding new stuff

> We could just add these 3 translations to the official excel file. This would be certainly the easiest option.

Let's do this later

**Comment by boldtrn** [comment]
> If it is not possible to use the same instance, we can still use the same classes avoiding rewriting later and adding new stuff

Yes, good point, that should work indeed. I will have a look and update this PR :)

File: mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResponseConverter.java [hunk 1]
```diff
@@ -244,9 +247,16 @@ private static void putBannerInstruction(InstructionList instructions, double di
...
         String bannerInstructionName = nextInstruction.getName();
-        if (bannerInstructionName == null || bannerInstructionName.isEmpty())
+        if (bannerInstructionName == null || bannerInstructionName.isEmpty()) {
             // Fix for final instruction and for instructions without name
             bannerInstructionName = nextInstruction.getTurnDescription(translationMap.getWithFallBack(locale));
+
+            // Uppercase first letter
+            // TODO: should we do this for all cases? Then we might change the spelling of street names though
+            if (bannerInstructionName != null && bannerInstructionName.length() > 1)
+                bannerInstructionName = bannerInstructionName.substring(0, 1).toUpperCase() + bannerInstructionName.substring(1);
```
Comment (devemux86) [inline mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResponseConverter.java]: Could use `toUpperCase(Locale.ROOT)`?

Comment (karussell) [reply mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResponseConverter.java]: Best would be Helper.toUpperCase or the Helper.firstBig for the whole but not that important :)

Comment (boldtrn) [reply mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResponseConverter.java]: Thanks for the feedback, I completely forgot about the helper method, I simply use the Helper method then :).

**Comment by boldtrn** [comment]
I updated this PR using as much as possible from the GH TranslationMap, I had to overwrite the doImport() method and needed to skip the `postImportHook()` call because this method is private. If think, at least right now, skipping the postImportHook is not an issue, if we have missing translations, this might become an issue?

File: mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResource.java [hunk 2]
```diff
@@ -53,11 +53,14 @@
...
     private final GraphHopperAPI graphHopper;
     private final TranslationMap translationMap;
+    private final MapboxResponseConverterTranslationMap mapboxResponseConverterTranslationMap;
 
     @Inject
     public MapboxResource(GraphHopperAPI graphHopper, TranslationMap translationMap) {
         this.graphHopper = graphHopper;
         this.translationMap = translationMap;
+        mapboxResponseConverterTranslationMap = new MapboxResponseConverterTranslationMap();
+        mapboxResponseConverterTranslationMap.doImport();
```
Comment (karussell) [inline mapbox/src/main/java/com/graphhopper/navigation/mapbox/MapboxResource.java]: We should make this static

**Comment by karussell** [comment]
I would not enforce the instance as singleton, just do 

static TranslationMap map = new TranslationMap().doImport();

in MapboxResource.java or alternatively inject a singleton into the MapboxResource (which would be even cleaner)

**Comment by boldtrn** [comment]
> inject a singleton into the MapboxResource (which would be even cleaner)

Ok sure, we can do this, I like that approach. But the injection would be out of scope of this project and would have to be done on the actual Server side, like gh-web. Should we change it that way?

**Comment by karussell** [comment]
Hmmh, good question. Is it possible that the MapboxResponseConverterTranslationMap does everything plus the new translations? Then we can replace this (on the server side)

bindFactory(TranslationMapFactory.class).to(TranslationMap.class);

with just

bindFactory(TranslationMapFactory.class).to(MapboxResponseConverterTranslationMap.class);

Everything else on the server side would be too ugly.

**Comment by karussell** [comment]
Ah, no. Forget my comment. It is the opposite: we need a custom TranslationMapFactory:

bindFactory(CustomTranslationMapFactory.class).to(TranslationMap.class);

but then e.g. the TranslationMap in GraphHopper is not swapped, which is ugly.

**Comment by karussell** [comment]
So, maybe keep it simple with the static variable in the MapboxResource

**Comment by boldtrn** [comment]
> So, maybe we should keep it simple with the static variable in the MapboxResource?

Ok sure, I can do that :)

**Comment by boldtrn** [comment]
Ok, I updated the PR again :)

**Pull request merged**